readthedocs/proxito/tests/test_redirects.py

```diff
@@ -55,6 +55,57 @@ def test_single_version_subproject_root_url_no_slash(self):
             r['Location'], 'https://project.dev.readthedocs.io/projects/subproject/',
         )
 
+    def test_subproject_redirect(self):
+        r = self.client.get('/', HTTP_HOST='subproject.dev.readthedocs.io')
```

Comment:
Does this break with the canonicalization around http -> https not doing the redirect?

Reply:
do you mean `http://suproject.rtd.io/foo`? well, we do http -> https redirects from cloudflare/nginx, so it gets redirected to `https://suproject.rtd.io/foo`. We don't support custom domains on the domain of the subproject, but if a user sets one, we would redirect directly to the main domain, instead of passing for https first, yeah. And we don't allow redirects from subprojects (well we do, but they don't work https://github.com/readthedocs/readthedocs.org/issues/5889).

I may start adding some validation in our forms to avoid having to deal with that at all p: